gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/JdbcExtractor.java

```diff
@@ -627,6 +630,10 @@ private void parseInputQuery(String query) {
       resultSet = statement.getResultSet();
     } catch (Exception e) {
       log.error("Failed to execute sql:" + query + " ;error-" + e.getMessage(), e);
+    } finally {
+      if (connection != null) {
```

Comment:
Can you catch the SQLException and add a log during closing, instead of throw it?